packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -725,14 +725,24 @@ class RenderParagraph extends RenderBox with ContainerRenderObjectMixin<RenderBo
   @override
   bool hitTestChildren(BoxHitTestResult result, { required Offset position }) {
     final TextPosition textPosition = _textPainter.getPositionForOffset(position);
-    final Object? span = _textPainter.text!.getSpanForPosition(textPosition);
-    if (span is HitTestTarget) {
-      result.add(HitTestEntry(span));
-      return true;
+    if (_positionIsWithinText(position: position, textPosition: textPosition)) {
+      final Object? span = _textPainter.text!.getSpanForPosition(textPosition);
+      if (span is HitTestTarget) {
+        result.add(HitTestEntry(span));
+        return true;
+      }
     }
     return hitTestInlineChildren(result, position);
   }
 
+  bool _positionIsWithinText({ required Offset position, required TextPosition textPosition }){
+    final List<ui.TextBox> boxes = _textPainter.getBoxesForRange(
+      textPosition.offset == 0 ? 0 : textPosition.offset - 1,
```

Comment:
If we really want to be certain, this range should be coming from the `span`, since this is where the hittest will be sent to

Reply:
There doesn't seem to be an existing api to get the range of a span in text painter.

I think boxes for (textPosition.offset - 1, textPosition.offset +1) should cover all cases.

Reply:
If there is no easy way to do that, I am fine with using textPosition.offset +-1

Also this can be 
max(0, textPosition.offset - 1)

min(_textPainter.text.length, textPosition.offset + 1)